keyboards/keebwerk/nano_slider/keymaps/midi2vol/keymap.c

```diff
@@ -0,0 +1,159 @@
+/* Copyright 2020 Duckle, 2021 Jesús Vallejo
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include QMK_KEYBOARD_H
+#include "analog.h"
+#include "qmk_midi.h"
+
+/*
+This keymap works with midi2vol: linux version  https://github.com/jesusvallejo/Midi2Vol-Linux  
+and windows version https://github.com/jesusvallejo/Midi2Vol
+
+Enables the user to change the volume on the host computer if running midi2vol software.
+
+Example to extend usability is provided check VLC in: enum custom_keycodes {} and bool process_record_user(uint16_t keycode, keyrecord_t *record) {}
+Remember to give allways the new volume control a hex value not used by other volume control, 
+and add it to the midi2vol software.
+Check respective midi2vol readme on how to do so.In linux change config.json,in windows you can use configuration option.
+
+A compiled version of this keymap is provided in here: https://github.com/jesusvallejo/nanokeymaps/
+
+*/
+
+uint8_t midi2vol = 0x3E;
+
+/* Defines names for use in layer keycodes and the keymap */
+
+#define _MEDIA 0    /* Controls Pause, Mute , Forward ... */
+
+#define _NAV 1      /* Nav arrows, Enter, Space*/
+
+#define _VOLUME 2   /* Changes midi2vol int to interface with midi2vol program: Chrome Volume, General Volume */
+
+#define _DISCORD 3  /* FXX unsused keys to interface with Discord: Mute , Silence */
+
+#define _LIGHTS 4   /* Edits underglow and retroilumination */
+
+#define _EDIT 5     /* Cut, Copy ,Paste */
+
+#define _RESET 6    /* Layer to set nano in bootloader mode */
+
+#define _TOOGLE 7   /* Momentary layer to switch between layers */
```

Comment:
We heavily recommend using enums, as they're easier to maintain long term, and less likely to have empty layers that artificially inflate the firmware size.
```suggestion
enum custom_layers {
    _MEDIA,    /* Controls Pause, Mute , Forward ... */
    _NAV,      /* Nav arrows, Enter, Space*/
    _VOLUME    /* Changes midi2vol int to interface with midi2vol program: Chrome Volume, General Volume */
    _DISCORD,  /* FXX unsused keys to interface with Discord: Mute , Silence */
    _LIGHTS,   /* Edits underglow and retroilumination */
    _EDIT,     /* Cut, Copy ,Paste */
    _RESET,    /* Layer to set nano in bootloader mode */
    _TOOGLE,   /* Momentary layer to switch between layers */
};
```